Regression in nightly-2020-09-23: "missing `as_flags` in implementation" of OpenOptionsExt

The following code compiles fine with nightly-2020-09-22 (and stable rustc since 1.10.0) but not nightly-2020-09-23.

```rust
use std::os::unix::fs::OpenOptionsExt;

struct MyOpenOptions;

impl OpenOptionsExt for MyOpenOptions {
    fn mode(&mut self, _mode: u32) -> &mut Self {
        unimplemented!()
    }

    fn custom_flags(&mut self, _flags: i32) -> &mut Self {
        unimplemented!()
    }
}
```

```console
error[E0046]: not all trait items implemented, missing: `as_flags`
 --> src/lib.rs:5:1
  |
5 | impl OpenOptionsExt for MyOpenOptions {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `as_flags` in implementation
  |
  = help: implement the missing item: `fn as_flags(&self) -> std::result::Result<i32, std::io::Error> { todo!() }`
```

This is minimized from https://github.com/async-rs/async-std/blob/e8126633a89aafea23259eb9faddb70b89f94423/src/fs/open_options.rs#L303-L313.

Mentioning #76110, #76801, @FedericoPonzi, @joshtriplett.

--- a/library/std/src/sys/unix/ext/fs.rs
+++ b/library/std/src/sys/unix/ext/fs.rs
@@ -348,31 +348,6 @@ pub trait OpenOptionsExt {
     /// ```
     #[stable(feature = "open_options_ext", since = "1.10.0")]
     fn custom_flags(&mut self, flags: i32) -> &mut Self;
-
-    /// Get the flags as [`libc::c_int`].
-    ///
-    /// This method allows the reuse of the OpenOptions as flags argument for [`libc::open`].
-    ///
-    /// [`libc::c_int`]: https://docs.rs/libc/*/libc/type.c_int.html
-    /// [`libc::open`]: https://docs.rs/libc/*/libc/fn.open.html
-    ///
-    /// # Examples
-    ///
-    /// ```no_run
-    /// # #![feature(rustc_private)]
-    /// #![feature(open_options_ext_as_flags)]
-    /// extern crate libc;
-    /// use std::ffi::CString;
-    /// use std::fs::OpenOptions;
-    /// use std::os::unix::fs::OpenOptionsExt;
-    ///
-    /// let mut options = OpenOptions::new();
-    /// options.write(true).read(true);
-    /// let file_name = CString::new("foo.txt").unwrap();
-    /// let file = unsafe { libc::open(file_name.as_c_str().as_ptr(), options.as_flags().unwrap()) };
-    /// ```
-    #[unstable(feature = "open_options_ext_as_flags", issue = "76801")]
-    fn as_flags(&self) -> io::Result<libc::c_int>;
 }
 
 #[stable(feature = "fs_ext", since = "1.1.0")]
@@ -385,10 +360,6 @@ impl OpenOptionsExt for OpenOptions {
     fn custom_flags(&mut self, flags: i32) -> &mut OpenOptions {
         self.as_inner_mut().custom_flags(flags);
         self
-    }
-
-    fn as_flags(&self) -> io::Result<libc::c_int> {
-        self.as_inner().as_flags()
     }
 }
 

--- a/library/std/src/sys/unix/fs.rs
+++ b/library/std/src/sys/unix/fs.rs
@@ -656,12 +656,6 @@ impl OpenOptions {
         self.mode = mode as mode_t;
     }
 
-    pub fn as_flags(&self) -> io::Result<c_int> {
-        let access_mode = self.get_access_mode()?;
-        let creation_mode = self.get_creation_mode()?;
-        Ok(creation_mode | access_mode | self.custom_flags)
-    }
-
     fn get_access_mode(&self) -> io::Result<c_int> {
         match (self.read, self.write, self.append) {
             (true, false, false) => Ok(libc::O_RDONLY),

